File: Flight-Reservation-System-Frontend/src/services/ManageReservationService.js
import axios from "axios";

const MANAGE_RESERVATION_API_BASE_URL = "http://localhost:8080/manageBooking";

class ManageReservationService {
  getAllReservation(userId) {
    return axios.get(
      MANAGE_RESERVATION_API_BASE_URL + "/viewAllReservation/" + userId
    );
  }

  getAllReservationBasedOnEmail(lastName, emailId) {
    return axios.get(
      MANAGE_RESERVATION_API_BASE_URL +
        "/viewAllReservation/" +
        lastName +
        "/" +
        emailId
    );
  }

  getOneReservation(bookingId, userId) {
    return axios.get(
      MANAGE_RESERVATION_API_BASE_URL +
        "/viewSingleReservation/" +
        bookingId +
        "/" +
        userId
    );
  }

  updateReservation(bookingId, userId, updatedReservation) {
    return axios.put(
      MANAGE_RESERVATION_API_BASE_URL +
        "/updateReservation/" +
        bookingId +
        "/" +
        userId,
      updatedReservation
    );
  }

  deleteReservation(bookingId, userId) {
    return axios.delete(
      MANAGE_RESERVATION_API_BASE_URL +
        "/deleteReservation/" +
        bookingId +
        "/" +
        userId
    );
  }
}

export default new ManageReservationService();

File: Flight-Reservation-System-Backend/src/main/java/com/flightreservation/service/ManageReservationService.java
package com.flightreservation.service;

import java.util.List;

import com.flightreservation.dto.ReservationDto;

public interface ManageReservationService {

	List<ReservationDto> getAllReservation(int userId);

	ReservationDto getOneReservation(int bookingId, int userId);

	List<ReservationDto> getAllReservationBasedOnEmail(String lastName, String emailId);

	ReservationDto updateReservation(ReservationDto updatedReservation, int bookingId, int userId);

	String deleteReservation(int bookingId, int userId);

	List<ReservationDto> getAllReservationBasedOnFlightNumber(String flightNumber);

}
